### ISSUE TITLE
Question : GNMT model faster on single GPU than on 4 GPUs?

### ISSUE THREAD

**Issue by pksubbarao** [NVIDIA/OpenSeq2Seq#191]
Hello
I have been experimenting with en-de-gnmt-like-4GPUs.py recipe. I have noticed that time per step is smaller if you use 1 GPU instead of 4 GPUs. It looks like step time is approximately 40% slower on 4 GPUs than 1 GPU. I have the default batch-size(32) in both cases.

Here is an example.
After training ~36 hours with 4 GPUs:
Global step 73968: loss = 1.0840, time per step = 0:00:1.390

However, after training ~36 hours with 1 GPU:
Global step 138432: loss = 1.2411, time per step = 0:00:0.789

I have a few questions:
1) Isn't the step time expected to be smaller on 4 GPU case?
2) However, after 36 hours, loss is smaller with 4 GPU case even though step time is higher. 
3) I'm wondering if that's because objects per second is higher on multi-GPU case. But if that is the case, shouldn't the step time reduce?

Thanks for clarification.

**Comment by billy19murahmi** [comment]
@pksubbarao when comparing single node vs multi, don't compare the step time.
Try to calculate the throughput, (global batch size / step time). This would help to find the number of sentences per second.

Example: 
Batch size per GPU = 512
No of GPU's = 4
Global batch size = 512 * 4 = 2048
step time = 1.39s
No of sentences/second = (2048/1.39) =  1473

where as in Single GPU,
batch size = 512
step time = 0.78
Hence, it would be, (512/0.78) = 656 Sentences/second

As you can see, Multi GPU throughput is faster.

Thanks,
Srini

**Comment by okuchaiev** [comment]
@billy19murahmi is correct:
total batch size = num_gpus (or workers in case of Horovod) * batch_size_per_gpu

So when you use more GPUs, your step time can go up, but your total_batch_size increases and you need less steps to go through your dataset

**Comment by pksubbarao** [comment]
Thanks for clarifying.
